### PR TITLE
feat: setup ETL proces to import GeoPackage into search index

### PR DESCRIPTION
**Pull request by rkettelerij**
# Description

Setup ETL proces to import a file (currently only gpkg is supported as source file type) into a search index (currently only postgresql is support as target search index). But the ETL proces is agnostic to a specific source/target.

Optimized for performance by loading records in batches and using COPY to insert the records in Postgres.

How to use:
```
NAME:
   gomagpie import-file - Import file into search index

USAGE:
   gomagpie import-file [command options]

CATEGORY:
   etl

OPTIONS:
   --db-host value        (default: "localhost") [$DB_HOST]
   --db-port value        (default: 5432) [$DB_PORT]
   --db-name value        Connect to this database [$DB_NAME]
   --db-username value    (default: "postgres") [$DB_USERNAME]
   --db-password value    (default: "postgres") [$DB_PASSWORD]
   --db-ssl-mode value    (default: "disable") [$DB_SSL_MODE]
   --config-file value    reference to YAML configuration file [$CONFIG_FILE]
   --search-index value   Name of search index in which to import the given file (default: "search_index") [$SEARCH_INDEX]
   --file value           Path to (e.g GeoPackage) file to import [$FILE]
   --fid value            Name of feature ID field in file (default: "fid") [$FID]
   --geom value           Name of geometry field in file (default: "geom") [$GEOM]
   --feature-table value  Name of the table in given file to import [$FEATURE_TABLE]
   --page-size value      Page/batch size to use when extracting records from file (default: 10000) [$PAGE_SIZE]
   --help, -h             show help
```

Example config file: 
```
---
version: 1.0.0
lastUpdated: "2024-10-22T12:00:00Z"
baseUrl: http://localhost:8080
availableLanguages:
  - nl
collections:
  - id: addresses
    metadata:
      title: Addresses
    search:
      fields:
        - foo
        - bar
      displayNameTemplate: "{{ .bar }} - {{ .foo }}"
      suggestTemplates:
        - "{{ .foo }} {{ .bar }}"
        - "{{ .bar }}, {{ .bar }} {{ .foo }}"
  - id: buildings
```

## Type of change

- New feature

# Checklist:

- [x] I've double-checked the code in this PR myself
- [x] I've left the code better than before ([boy scout rule](https://wiki.c2.com/?BoyScoutRule))
- [x] The code is readable, comments are added that explain hard or non-obvious parts.
- [x] I've expanded/improved the (unit) tests, when applicable
- [x] I've run (unit) tests that prove my solution works
- [x] There's no sensitive information like credentials in my PR

### CODE REVIEW

File: internal/etl/extract/geopackage.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func (g *GeoPackage) Extract(table config.FeatureTable, fields []string, limit int, offset int) ([]t.RawRecord, error) {
+	if len(fields) == 0 {
+		return nil, errors.New("no fields provided to read from GeoPackage")
+	}
+
+	query := fmt.Sprintf(`
+		select %[3]s as fid,
+		    st_minx(castautomagic(%[4]s)) as bbox_minx,
```
Comment (roelarents) [inline internal/etl/extract/geopackage.go:66]: je kunt `enablegpkgmode()` overwegen
https://kadaster-it.slack.com/archives/C02T4HWN5CZ/p1702376762334029?thread_ts=1702375792.264109&cid=C02T4HWN5CZ

File: internal/etl/transform/transform.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		fieldValuesByName, err := slicesToMap(collection.Search.Fields, r.FieldValues)
+		if err != nil {
+			return nil, err
+		}
+		displayName, err := t.renderTemplate(collection.Search.DisplayNameTemplate, fieldValuesByName)
+		if err != nil {
+			return nil, err
+		}
+		suggestions := make([]string, 0, len(collection.Search.SuggestTemplates))
+		for _, suggestTemplate := range collection.Search.SuggestTemplates {
```
Comment (roelarents) [inline internal/etl/transform/transform.go:47]: de substitutions en alternatieve spellingen gaan geregeld worden in de zoeker, right?

File: internal/etl/load/postgres.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	}
+	return &Postgis{db: db, ctx: ctx}, nil
+}
+
+func (p *Postgis) Close() {
+	_ = p.db.Close(p.ctx)
+}
+
+func (p *Postgis) Load(records []t.SearchIndexRecord, index string) (int64, error) {
+	loaded, err := p.db.CopyFrom(
```
Comment (roelarents) [inline internal/etl/load/postgres.go:35]: 👍

**Review by roelarents — APPROVED** [review]

File: internal/etl/extract/geopackage.go (hunk 1 above)
Comment (rkettelerij) [reply internal/etl/extract/geopackage.go:67]: check. Heb het nu even bij die casts gehouden. Lijkt verder geen verschil te maken. `enablegpkgmode` maakt het overigens nog steeds niet mogelijk om een st_transform te doen, zie https://gis.stackexchange.com/q/412884. Dat gaan we misschien ook nog nodig hebben (nu een todo gezet).

File: config/collections.go [hunk 4]
```diff
@@ -12,13 +12,16 @@ type GeoSpatialCollections []GeoSpatialCollection
 
 type GeoSpatialCollection struct {
 	// Unique ID of the collection
-	ID string `yaml:"id" validate:"required" json:"id"`
+	ID string `yaml:"id" json:"id" validate:"required" `
```
[the remark this answers is not in the document]
Comment (rkettelerij) [reply config/collections.go]: fixed

**Pull request merged**